File: TS/JSONRPCResponse.ts
/**
 * A successful (non-error) response to a request.
 */
import { RequestId } from './RequestId';
import { Result } from './Result';
import { JSONRPC_VERSION } from './constants';

export interface JSONRPCResponse {
  jsonrpc: typeof JSONRPC_VERSION;
  id: RequestId;
  result: Result;
}

File: Sources/Schema/JSONRPCResponse.swift
/**
 * A successful (non-error) response to a request.
 */
public struct JSONRPCResponse: Codable {
    public var jsonrpc: String = Constants.JSONRPC_VERSION
    public var id: RequestId
    public var result: AnyCodable
    
    public init(id: RequestId, result: AnyCodable) {
        self.id = id
        self.result = result
    }
}
